File: 1-two-sum/two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        # hasmap
        num_map = {}
        for i , n in enumerate(nums):
            diff = target - n
            if diff in num_map:
                return[num_map[diff], i]
            num_map[n] = i
        return None

File: 1-two-sum/two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    const hashM = {};
    for (let i = 0; i<nums.length; i++){
        const num = nums[i]
        if(target - num in hashM){
            return [i, hashM[target - num]];
        }
        hashM[num] = i;
    }
};